core/src/Kokkos_HIP_Space.hpp

```diff
@@ -0,0 +1,633 @@
+// modification, are permitted provided that the following conditions are
+// met:
+//
+// 1. Redistributions of source code must retain the above copyright
+// notice, this list of conditions and the following disclaimer.
+//
+// 2. Redistributions in binary form must reproduce the above copyright
+// notice, this list of conditions and the following disclaimer in the
+// documentation and/or other materials provided with the distribution.
+//
+// 3. Neither the name of the Corporation nor the names of the
+// contributors may be used to endorse or promote products derived from
+// this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY SANDIA CORPORATION "AS IS" AND ANY
+// EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+// PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL SANDIA CORPORATION OR THE
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+// PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+// LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+// NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+// SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// Questions? Contact Christian R. Trott (crtrott@sandia.gov)
+//
+// ************************************************************************
+//@HEADER
+*/
+
+#ifndef KOKKOS_HIPSPACE_HPP
+#define KOKKOS_HIPSPACE_HPP
+
+#include <Kokkos_Core_fwd.hpp>
+
+#if defined(KOKKOS_ENABLE_HIP)
+
+#include <iosfwd>
+#include <typeinfo>
+#include <string>
+
+#include <Kokkos_HostSpace.hpp>
+
+#include <hip/hip_runtime_api.h>
+/*--------------------------------------------------------------------------*/
+
+namespace Kokkos {
+namespace Experimental {
+/** \brief  HIP on-device memory management */
+
+class HIPSpace {
+ public:
+  //! Tag this class as a kokkos memory space
+  typedef HIPSpace memory_space;
+  typedef Kokkos::Experimental::HIP execution_space;
+  typedef Kokkos::Device<execution_space, memory_space> device_type;
+
+  typedef unsigned int size_type;
+
+  /*--------------------------------*/
+
+  HIPSpace();
+  HIPSpace(HIPSpace&& rhs)      = default;
+  HIPSpace(const HIPSpace& rhs) = default;
+  HIPSpace& operator=(HIPSpace&& rhs) = default;
+  HIPSpace& operator=(const HIPSpace& rhs) = default;
+  ~HIPSpace()                              = default;
+
+  /**\brief  Allocate untracked memory in the rocm space */
+  void* allocate(const size_t arg_alloc_size) const;
+
+  /**\brief  Deallocate untracked memory in the rocm space */
+  void deallocate(void* const arg_alloc_ptr, const size_t arg_alloc_size) const;
+
+  /**\brief Return Name of the MemorySpace */
+  static constexpr const char* name() { return m_name; };
```

Comment:
Haha: someone else complained about this on my OpenMPTarget backend: just return the string directly here and delete the member :-D

Reply:
That was good advice